lambda/matching.ml

```diff
@@ -3720,18 +3683,8 @@ let do_for_multiple_match ~scopes loc paraml pat_act_list partial =
           comp_match_handlers (compile_flattened ~scopes repr) partial
             (Context.start size) flat_next flat_nexts
         in
-        List.fold_right2 (bind Strict) idl paraml
-          ( match partial with
-          | Partial ->
-              check_total ~scopes loc ~failer:Raise_match_failure
-                total lam raise_num
-          | Total ->
-              assert (Jumps.is_empty total);
-              lam
-          )
-  with Unused -> assert false
-
-(* ; partial_function loc () *)
+        List.fold_right2 (bind Strict) idl paraml lam, total
```

Comment:
This moves the bindings below the handler for the match failure. However, making variables more local seems good.